universal-login-sdk/test/integration/core/BalanceObserver.ts

```diff
@@ -2,64 +2,128 @@ import {expect} from 'chai';
 import sinon from 'sinon';
 import {utils, Wallet, providers, Contract} from 'ethers';
 import {deployContract, createMockProvider, getWallets} from 'ethereum-waffle';
-import {ETHER_NATIVE_TOKEN, TEST_ACCOUNT_ADDRESS, sleep, waitUntil} from '@universal-login/commons';
+import {ETHER_NATIVE_TOKEN, TEST_ACCOUNT_ADDRESS, waitUntil, BalanceChecker, TokenDetails} from '@universal-login/commons';
 import MockToken from '@universal-login/contracts/build/MockToken.json';
 import {BalanceObserver} from '../../../lib/core/observers/BalanceObserver';
 
 describe('INT: BalanceObserver', () => {
+  let balanceChecker: BalanceChecker;
   let balanceObserver: BalanceObserver;
   let provider: providers.Provider;
   let wallet: Wallet;
   let mockToken: Contract;
-  const minimalAmount =  utils.parseEther('0.5').toString();
-  let supportedTokens = [
-    {
-      address: ETHER_NATIVE_TOKEN.address,
-      minimalAmount
-    }
-  ];
-  let callback: sinon.SinonSpy;
-  let unsubscribe: any;
-
-
-  beforeEach(async () => {
-    provider = createMockProvider();
-    [wallet] = getWallets(provider);
-    mockToken = await deployContract(wallet, MockToken);
-    supportedTokens = [...supportedTokens, {address: mockToken.address, minimalAmount}];
-    callback = sinon.spy();
-    balanceObserver = new BalanceObserver(supportedTokens, provider);
-    balanceObserver.step = 10;
-    unsubscribe = await balanceObserver.startAndSubscribe(TEST_ACCOUNT_ADDRESS, callback);
-  });
 
-  it('calls callback if ether balance changed', async () => {
-    await wallet.sendTransaction({to: TEST_ACCOUNT_ADDRESS, value: utils.parseEther('1.0')});
-    await waitUntil(() => !!callback.firstCall);
-    expect(callback).to.have.been.calledWith(ETHER_NATIVE_TOKEN.address, TEST_ACCOUNT_ADDRESS);
-    unsubscribe();
-  });
+  describe('BalanceObserver', () => {
+    beforeEach(async () => {
+      provider = createMockProvider();
+      [wallet] = getWallets(provider);
+      mockToken = await deployContract(wallet, MockToken);
+      const supportedTokens: TokenDetails[] = [
+        {address: ETHER_NATIVE_TOKEN.address, symbol: 'ETH', name: 'Ethereum'},
```

Comment:
`{address: ETHER_NATIVE_TOKEN.address, symbol: 'ETH', name: 'Ethereum'}`
 -> 
`ETHER_NATIVE_TOKEN`

Reply:
Name of this currency is ether, no Ethereum.

Reply:
use [TokenService](https://github.com/UniversalLogin/UniversalLoginSDK/blob/master/universal-login-wallet/src/integration/ethereum/TokenService.ts) to get token details